erpnext/stock/doctype/pick_list/pick_list.py

```
@@ -1010,17 +1038,22 @@ def get_available_item_locations(
 	if not ignore_validation:
 		validate_picked_materials(item_code, required_qty, locations, picked_item_details)
 
 	return locations
 
 
-def get_locations_based_on_required_qty(locations, required_qty):
+def get_locations_based_on_required_qty(locations, required_qty, priority_warehouses):
 	filtered_locations = []
 
+	if priority_warehouses:
+		priority_locations = [l for l in locations if l.warehouse in priority_warehouses]
```

Ruff (0.15.7)

[error] Ambiguous variable name: `l`

(E741)